src/core/connections.jl

```diff
@@ -44,7 +44,16 @@ function _check_labels(md::ModelDef, comp_def::ComponentDef, param_name::Symbol,
     for (i, dim) in enumerate(comp_dims)
         if isa(dim, Symbol) 
             # if length(index_values[dim]) != size(ext_param.values)[i]
-            if dim_count(md, dim) != size(ext_param.values)[i]
+            if dim == :time 
+                first = comp_def.first 
+                last = comp_def.last 
+                t = md.dimensions[:time]
+                n1 = t[last] - t[first] + 1
+                n2 = size(ext_param.values)[i]
+                if n1 != n2
+                    error("Mismatched data size for a parameter connection: $(comp_def.comp_id): dim :time has $n1 elements; parameter :$param_name has $n2 elements.")
+                end
+            elseif dim_count(md, dim) != size(ext_param.values)[i]
```

Comment:
You can consolidate the redundant portions of this so that you capture n1, n2, and dim in both branches of the conditional, then after the conditional, test n1 != n2 and raise the error.